core/src/main/java/org/apache/calcite/rex/RexCall.java

```diff
@@ -127,11 +137,83 @@ protected final StringBuilder appendOperands(StringBuilder sb) {
           includeType = RexDigestIncludeType.NO_TYPE;
         }
       }
-      sb.append(((RexLiteral) operand).computeDigest(includeType));
+      operandDigests.add(((RexLiteral) operand).computeDigest(includeType));
+    }
+    int totalLength = (operandDigests.size() - 1) * 2; // commas
+    for (String s : operandDigests) {
+      totalLength += s.length();
+    }
+    sb.ensureCapacity(sb.length() + totalLength);
+    sortOperandsIfNeeded(sb, operands, operandDigests);
+    for (int i = 0; i < operandDigests.size(); i++) {
+      String op = operandDigests.get(i);
+      if (i != 0) {
+        sb.append(", ");
+      }
+      sb.append(op);
     }
     return sb;
   }
 
+  private void sortOperandsIfNeeded(StringBuilder sb,
+      List<RexNode> operands, List<String> operandDigests) {
+    if (operands.isEmpty() || !needNormalize()) {
+      return;
+    }
+    final SqlKind kind = op.getKind();
+    if (SqlKind.SYMMETRICAL_SAME_ARG_TYPE.contains(kind)) {
+      final RelDataType firstType = operands.get(0).getType();
+      for (int i = 1; i < operands.size(); i++) {
+        if (!equalSansNullability(firstType, operands.get(i).getType())) {
+          // Arguments have different type, thus they must not be sorted
+          return;
+        }
+      }
+      // fall through: order arguments below
+    } else if (!SqlKind.SYMMETRICAL.contains(kind)
+        && (kind == kind.reverse()
```

Comment:
I have left the comments in another PR https://github.com/apache/calcite/pull/1740